addons/account/models/res_partner_bank.py

```diff
@@ -32,6 +52,56 @@ def _check_journal_id(self):
             if len(bank.journal_id) > 1:
                 raise ValidationError(_('A bank account can belong to only one journal.'))
 
+    @api.constrains('allow_out_payment')
+    def _check_allow_out_payment(self):
+        """ Block enabling the setting, but it can be set to false without the group. (For example, at creation) """
+        for bank in self:
+            if bank.allow_out_payment:
+                if not self.user_has_groups('account.group_validate_bank_account'):
+                    raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
+
+    @api.depends('partner_id', 'sanitized_acc_number', 'allow_out_payment')
+    def _compute_display_account_warning(self):
+        for bank in self:
+            if bank.allow_out_payment or not bank.sanitized_acc_number or bank.acc_type != 'iban':
+                bank.iban_warning = False
+                bank.money_transfer_warning = False
+                continue
+            bank_country = bank.sanitized_acc_number[:2]
+            if bank_country != bank.partner_id.country_id.code:
+                bank.iban_warning = markupsafe.Markup("""
+                    <span class="text-warning">Medium risk</span>: Iban <span class="fw-bold">{}</span>
```

Comment:
translations ? (and translating means not storing it obviously)

Reply:
Could it work with `@api.depends_context('lang')`? (with translations, indeed it slipped my mind)

Putting it in the view means we need another field for the money transfer institution. (Or forget about displaying it, but the PO seemed to really want it).

Reply:
We can simply don't store those fields.

Reply:
@KangOl I'm now storing bool for whether or not the warning is there (which still matches the spec as we need to use them as a filter). As for the messages, they're still computed but no longer stored.
I adapted the upgrade pr too